### ISSUE TITLE
Run all solution scripts in CI

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [LearningTypeScript/projects#2]
Projects should have a command like (or exactly? #1) `test:solutions`. These should be run in CI to validate solutions all work.

**Comment by JoshuaKGoldberg** [comment]
For now, I'm ok with all projects being switched to `test:solutions`.

**Comment by fuadop** [comment]
Hey @JoshuaKGoldberg 👋🏾 .
If you were to take on this? How would you go about it ? 
Since each project has it's own package.json and test:solutions script in it ? 👀

**Comment by fuadop** [comment]
Or do I write a script that spawns a child process ? like it was done in this stackoverflow solution ? 
![Screenshot from 2022-05-24 21-01-14](https://user-images.githubusercontent.com/65264054/170122132-1f26a3b1-e8d6-4298-a3d5-f57ed80f127e.png)

**Comment by JoshuaKGoldberg** [comment]
Hmm, great question. I'd prefer we didn't write any Node scripts to wrangle tests. Those tend to be hard to maintain. This should be solveable just with some magic combination of `npm run` scripts, GitHub Actions (GHA), [Jest projects](https://jestjs.io/docs/configuration#projects-arraystring--projectconfig), and/or [TypeScript project references](https://www.typescriptlang.org/docs/handbook/project-references.html).

Another interesting thing to note: there are going to be [_dozens_ of projects](https://www.youtube.com/watch?v=lKie-vgUGdI). Keeping a big list in a file of all the projects to include would be an annoying task, potentially a lot of code, and something that could get out of date easily (unless there's _even more_ code written to verify it's up to date 🤢). So it would probably be much better to use `*` asterisk/wildcard-based file searching systems to find all the projects, if possible.

The quickest way to get this to work might be some script in GHA that loops over all the child directories except `node_modules/` and runs `npm run test:solutions` in it [[StackOverflow starter for looping over directories](https://unix.stackexchange.com/questions/86722/how-do-i-loop-through-only-directories-in-bash)]. This would run them in series and would need to be refactored later to run them in parallel - but since there are only a few projects right now, that can always be a followup issue. [MVP approach, baby!](https://en.wikipedia.org/wiki/Minimum_viable_product)
